### ISSUE TITLE
Add missing negative/edge-case test coverage for thin analyzers (Moq1100, Moq1200, Moq1202, Moq1300, Moq1400, Moq1420)

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#1270]
## Summary

An internal code audit ranked the analyzer test suites by negative/edge-case coverage. The test infrastructure itself is strong (markup like `{|Moq1002:(1, true)|}` asserts both diagnostic ID and exact span, unexpected diagnostics fail the test, and `TestDataExtensions` multiplies every row across 2 namespace styles x 2 Moq versions), but six analyzers have thin data sets, two test fixtures contain **dead members that mask untested scenarios**, and malformed/incomplete-code tests exist in only 2 of 10 sampled test classes.

This issue adds the missing rows. **Every expected diagnostic below was verified by running the current analyzers against the exact test code** — this issue adds coverage that pins *current* behavior; it changes no analyzer. Where a row documents a known false negative, a comment in the row says so, so the row self-explains when a future fix flips it.

**Deliberately excluded** (they belong to companion audit issues that change analyzer behavior, and tests must land with the behavior change):

- Moq1300 on *error types* and *generic type parameters* (`mock.As<T>()` in a generic method): the companion issue "Moq1300 fires at Error severity on unresolved/type-parameter type arguments" changes this behavior; its tests land there.
- Moq1202/Moq1204 rows for non-generic `System.EventHandler` events and custom `(object, EventArgs)` delegates: covered by the companion issue on the `Raise`/`Raises` EventHandler false positive.
- Moq1400 rows passing `MockBehavior` through a local variable or `const`: covered by the companion issue on boxed-enum reference equality in `SetExplicitMockBehaviorAnalyzer`/`SetStrictMockBehaviorAnalyzer`.

## Current state

All paths relative to repo root; all line numbers verified against the current `main`.

| Analyzer | Test file | Gap |
|---|---|---|
| Moq1300 | `tests/Moq.Analyzers.Test/AsAcceptOnlyInterfaceAnalyzerTests.cs:11-14` | Only 4 data rows; no generic/nested-generic interface, no chained `As<I1>().As<I2>()`, no `As` on a mock variable, no delegate/struct/enum type arguments |
| Moq1400 | `tests/Moq.Analyzers.Test/SetExplicitMockBehaviorAnalyzerTests.cs:10-34` | No ctor-args+behavior combinations, no `MockRepository.Create<T>()` rows, no `Mock.Of<T>(predicate)` rows, no malformed code |
| Moq1202 | `tests/Moq.Analyzers.Test/RaiseEventArgumentsShouldMatchEventSignatureAnalyzerTests.cs` | Only `Action`/`Action<T>`/`EventHandler<T>` single-arg events; no multi-parameter `Action<T1,T2>`, no nested-generic payload, no `null` argument, no `Raise` via member/method-chain mock access |
| Moq1420 | `tests/Moq.Analyzers.Test/RedundantTimesSpecificationAnalyzerTests.cs` | No `Func<Times>` method-group form (`Times.AtLeastOnce` without parentheses), no `using static Moq.Times`, no Times-in-local, no message-only `Verify` |
| Moq1200 | `tests/Moq.Analyzers.Test/SetupShouldBeUsedOnlyForOverridableMembersAnalyzerTests.cs:93` | Fixture class `SampleClassWithStaticMembers` is declared but referenced by **zero** data rows (dead fixture); no extension-method Setup, no nested-generic mocked type, no malformed lambda |
| Moq1100 | `tests/Moq.Analyzers.Test/CallbackSignatureShouldMatchMockedMethodAnalyzerTests.cs:123-126,130` | 5 fixture methods are declared and never used by any data row (dead fixtures): `ProcessData` (ref), `TryProcess` (out), `ProcessMixed` (int/ref/out), `ProcessReadOnly` (in), `ProcessGeneric<T>`; no method-group callback, no custom-delegate callback, no async-lambda callback |
| Moq1000 | `tests/Moq.Analyzers.Test/NoSealedClassMocksAnalyzerTests.cs:171` | Cosmetic: test name typo `ShoulAnalyzeSealedClassMocks` |

Dead fixture evidence (verbatim):

`tests/Moq.Analyzers.Test/SetupShouldBeUsedOnlyForOverridableMembersAnalyzerTests.cs:93`:

```csharp
public class SampleClassWithStaticMembers { public static int StaticField; public const int ConstField = 42; public static readonly int ReadonlyField = 42; public static void StaticMethod() { } }
```

`tests/Moq.Analyzers.Test/CallbackSignatureShouldMatchMockedMethodAnalyzerTests.cs:123-130` (`:127-129` — `DoRef`/`DoOut`/`DoIn` — *are* used; the other five are not):

```csharp
void ProcessData(ref string data);
bool TryProcess(out int result);
void ProcessMixed(int id, ref string data, out bool success);
void ProcessReadOnly(in DateTime timestamp);
int DoRef(ref string data);
bool DoOut(out int result);
string DoIn(in DateTime timestamp);
T ProcessGeneric<T>(T input);
```

Cross-cutting: malformed/incomplete-code tests exist in only 2 of 10 sampled classes, even though mid-edit robustness is the project's #1 priority. The existing pattern to copy is `CallbackSignatureShouldMatchMockedMethodAnalyzerTests.UnresolvableParameterType_ReportsDiagnostic` (`tests/Moq.Analyzers.Test/CallbackSignatureShouldMatchMockedMethodAnalyzerTests.cs:195-218`), which passes `CompilerDiagnostics.None` to `AnalyzerVerifier.VerifyAnalyzerAsync` (overload at `tests/Moq.Analyzers.Test/Helpers/AnalyzerVerifier.cs:35`).

Useful infrastructure facts (verified):

- `Test<TAnalyzer,TCodeFixProvider>` injects `global using System; global using System.Collections.Generic; global using System.Threading.Tasks; global using Moq;` into every test compilation (`tests/Moq.Analyzers.Test/Helpers/Test.cs:20-28`), so rows may use `IList<int>`, `List<string>`, `Task.Delay`, etc. without qualifying.
- `WithNamespaces()` prepends `""` and `"namespace MyNamespace;"` variants; `WithMoqReferenceAssemblyGroups()` runs each row against Moq 4.8.2 and 4.18.4 (`tests/Moq.Analyzers.Test/Helpers/TestDataExtensions.cs`).
- Both Moq 4.8.2 and 4.18.4 expose `Verify(expression, Func<Times>)`, `VerifyGet(..., Func<Times>)`, and `VerifySet(..., Func<Times>)` overloads (verified against both package DLLs), so the method-group rows below compile on both.
- Moq 4.8.2 already has `ICallback.Callback(Delegate)` (verified against the package DLL), so the ref/in/method-group/custom-delegate callback rows below compile on both versions.

## Desired state

Each of the six analyzers has data rows (or dedicated `[Theory]`/`[Fact]` tests for sources that need extra `using` directives or `CompilerDiagnostics.None`) for the scenarios listed below, all fixture members are referenced by at least one row, and every row's expectation matches current analyzer behavior. Rows that pin a known false negative carry a comment saying so.

### 1. Moq1300 — `tests/Moq.Analyzers.Test/AsAcceptOnlyInterfaceAnalyzerTests.cs`

1a. Add these rows to the `TestData()` array (after line 14):

```csharp
// Generic and nested-generic interface type arguments (should NOT be flagged)
["""new Mock<SampleClass>().As<IList<int>>();"""],
["""new Mock<SampleClass>().As<IGenericInterface<IList<int>>>();"""],
["""new Mock<SampleClass>().As<IDictionary<string, IList<int>>>();"""],

// Chained As calls, both interfaces (should NOT be flagged)
["""new Mock<SampleClass>().As<ISampleInterface>().As<IOtherInterface>();"""],

// As on a mock variable rather than an inline creation
["""var mock = new Mock<SampleClass>(); mock.As<ISampleInterface>();"""],
["""var mock = new Mock<SampleClass>(); mock.As<{|Moq1300:SampleClass|}>();"""],

// Delegate type arguments are not interfaces (flagged; message uses the display string)
["""new Mock<SampleClass>().As<{|Moq1300:SampleDelegate|}>();"""],
["""new Mock<SampleClass>().As<{|Moq1300:Action<int>|}>();"""],
```

1b. Extend the `Template` fixture (inside the raw string in `ShouldAnalyzeAs`, after the `ISampleInterface` declaration) with:

```csharp
public interface IOtherInterface
{
    void Do();
}

public interface IGenericInterface<T>
{
    T Get();
}

public delegate void SampleDelegate(int x);
```

1c. Struct/enum type arguments violate `As<TInterface>`'s `where TInterface : class` constraint (CS0452). Overload resolution fails, so the analyzer stays silent — verified. Pin that with a dedicated test (also add `using Microsoft.CodeAnalysis.Testing;` at the top of the file for `CompilerDiagnostics`):

```csharp
/// <summary>
/// As&lt;T&gt; has a 'where T : class' constraint, so struct/enum type arguments produce CS0452.
/// Overload resolution fails and the analyzer intentionally stays silent (current behavior).
/// </summary>
[Theory]
[InlineData("Net80WithOldMoq")]
[InlineData("Net80WithNewMoq")]
public async Task ShouldNotReportWhenTypeArgumentViolatesClassConstraint(string referenceAssemblyGroup)
{
    const string source = """
        public enum SampleEnum { A }
        public struct SampleStruct { public int X; }
        public class SampleClass { }

        internal class UnitTest
        {
            private void Test()
            {
                new Mock<SampleClass>().As<SampleStruct>();
                new Mock<SampleClass>().As<SampleEnum>();
            }
        }
        """;

    // CompilerDiagnostics.None suppresses CS0452 from the class-constraint violation.
    await Verifier.VerifyAnalyzerAsync(source, referenceAssemblyGroup, CompilerDiagnostics.None);
}
```

*Scope note:* error-type and type-parameter arguments are intentionally **not** covered here (see Summary — companion Moq1300 severity/error-type issue).

### 2. Moq1400 — `tests/Moq.Analyzers.Test/SetExplicitMockBehaviorAnalyzerTests.cs`

2a. Add to the `common` list (after line 21):

```csharp
// Constructor arguments with and without an explicit behavior
["""new Mock<Foo>(MockBehavior.Loose, 1, 2);"""],
["""{|Moq1400:new Mock<Foo>(1, 2)|};"""],
["""{|Moq1400:new Mock<Foo>(MockBehavior.Default, 1, 2)|};"""],

// MockRepository.Create<T>() is not analyzed today (no diagnostic, even when the
// behavior argument is MockBehavior.Default) - these rows pin the current behavior.
["""var repository = new MockRepository(MockBehavior.Strict); repository.Create<ISample>();"""],
["""var repository = new MockRepository(MockBehavior.Strict); repository.Create<ISample>(MockBehavior.Default);"""],
```

2b. Add to the `newMoqOnly` list (after line 31; kept in the new-Moq group because the two-argument `Mock.Of<T>(predicate, MockBehavior)` overload was added in Moq 4.12.0, matching the existing comment at line 24):

```csharp
// Mock.Of<T>(predicate) overloads
["""{|Moq1400:Mock.Of<ISample>(s => s.Name == "x")|};"""],
["""Mock.Of<ISample>(s => s.Name == "x", MockBehavior.Loose);"""],
["""{|Moq1400:Mock.Of<ISample>(s => s.Name == "x", MockBehavior.Default)|};"""],
```

2c. Extend the theory's inline fixture (the `ISample` interface in `ShouldAnalyzeExplicitMockBehavior`) to:

```csharp
public interface ISample
{
    void Method();
    string Name { get; set; }
}

public class Foo
{
    public Foo(int a, int b) { }
    public Foo(MockBehavior behavior, int a, int b) { }
}
```

2d. Add a malformed-code test (and `using Microsoft.CodeAnalysis.Testing;` at the top of the file). Note: the shorter fragment `new Mock<ISample>(` *does* produce a Moq1400 whose recovered span extends past the end of the line, which `{|...|}` markup cannot express cleanly, so this test uses the `MockBehavior.`-member-access fragment, which produces no diagnostic (verified):

```csharp
/// <summary>
/// Incomplete member access inside the creation argument must not crash the analyzer
/// and produces no diagnostic (current behavior).
/// </summary>
[Theory]
[InlineData("Net80WithOldMoq")]
[InlineData("Net80WithNewMoq")]
public async Task ShouldNotCrashOnIncompleteMockCreation(string referenceAssemblyGroup)
{
    const string source = """
        public interface ISample
        {
            void Method();
        }

        internal class UnitTest
        {
            private void Test()
            {
                new Mock<ISample>(MockBehavior.
            }
        }
        """;

    // CompilerDiagnostics.None suppresses CS1001/CS1026/CS1002/CS0117 from the incomplete code.
    await Verifier.VerifyAnalyzerAsync(source, referenceAssemblyGroup, CompilerDiagnostics.None);
}
```

*Scope note:* `MockBehavior` passed through a local/`const` is intentionally **not** covered here (see Summary — companion boxed-enum equality issue).

### 3. Moq1202 — `tests/Moq.Analyzers.Test/RaiseEventArgumentsShouldMatchEventSignatureAnalyzerTests.cs`

3a. Add to `ValidTestData()`:

```csharp
// Valid: Action<string, int> event with matching arguments
["""mockProvider.Raise(p => p.PairChanged += null, "key", 42);"""],

// Valid: nested generic payload Action<List<string>>
["""mockProvider.Raise(p => p.ListChanged += null, new List<string>());"""],

// Valid: null literal for a reference-type parameter. The cast is required:
// a bare 'null' is ambiguous between Raise(..., EventArgs) and Raise(..., params object[]) (CS0121).
["""mockProvider.Raise(p => p.StringOptionsChanged += null, (string)null);"""],
```

3b. Add to `InvalidTestData()`:

```csharp
// Invalid: wrong second argument for Action<string, int> (wrap only the problematic argument)
["""mockProvider.Raise(p => p.PairChanged += null, "key", {|Moq1202:"wrong"|});"""],

// Invalid: too few arguments for Action<string, int> (wrap the entire invocation)
["""{|Moq1202:mockProvider.Raise(p => p.PairChanged += null, "key")|};"""],

// Invalid: wrong generic payload - List<int> supplied to Action<List<string>>
["""mockProvider.Raise(p => p.ListChanged += null, {|Moq1202:new List<int>()|});"""],
```

3c. Add these two events to the `IOptionsProvider` interface in **both** the `ShouldNotReportDiagnosticForValidRaiseArguments` template (lines 69-76) and the `ShouldReportDiagnosticForInvalidRaiseArguments` template (lines 101-106):

```csharp
event Action<string, int> PairChanged;
event Action<List<string>> ListChanged;
```

3d. Add a new theory for `Raise` reached through member access and method chains:

```csharp
public static IEnumerable<object[]> MockAccessPatternTestData()
{
    return new object[][]
    {
        // Valid: Raise on a mock exposed as a property
        ["""wrapper.ProviderMock.Raise(p => p.StringOptionsChanged += null, "ok");"""],

        // Invalid: Raise on a mock exposed as a property
        ["""wrapper.ProviderMock.Raise(p => p.StringOptionsChanged += null, {|Moq1202:42|});"""],

        // Invalid: Raise on a mock returned from a method call
        ["""wrapper.GetMock().Raise(p => p.StringOptionsChanged += null, {|Moq1202:42|});"""],
    }.WithNamespaces().WithMoqReferenceAssemblyGroups();
}

[Theory]
[MemberData(nameof(MockAccessPatternTestData))]
public async Task ShouldAnalyzeRaiseOnMemberAndMethodChainMockAccess(string referenceAssemblyGroup, string @namespace, string raiseCall)
{
    await Verifier.VerifyAnalyzerAsync(
        $$"""
          {{@namespace}}

          internal interface IOptionsProvider
          {
              event Action<string> StringOptionsChanged;
          }

          internal class Wrapper
          {
              public Mock<IOptionsProvider> ProviderMock { get; } = new Mock<IOptionsProvider>();

              public Mock<IOptionsProvider> GetMock() => ProviderMock;
          }

          internal class UnitTest
          {
              private void Test()
              {
                  var wrapper = new Wrapper();
                  {{raiseCall}}
              }
          }
          """,
        referenceAssemblyGroup);
}
```

*Scope note:* non-generic `EventHandler` and custom-delegate events are intentionally **not** covered here (see Summary — companion Raise/Raises EventHandler issue).

### 4. Moq1420 — `tests/Moq.Analyzers.Test/RedundantTimesSpecificationAnalyzerTests.cs`

4a. Add to the `both` list (this file uses regular strings with escaped quotes; keep that style):

```csharp
// Should NOT flag the Func<Times> method-group form today: those overloads take
// Func<Times>, not Times, so the analyzer's Times-parameter check never matches.
// Both Moq 4.8.2 and 4.18.4 expose Verify/VerifyGet(..., Func<Times>) overloads.
// This is a known false negative; these rows pin the current behavior.
["new Mock<ISampleInterface>().Verify(x => x.TestMethod(), Times.AtLeastOnce);"],
["new Mock<ISampleInterface>().VerifyGet(x => x.TestProperty, Times.AtLeastOnce);"],

// Should NOT detect Times stored in a local variable (argument is not an invocation)
["var times = Times.AtLeastOnce(); new Mock<ISampleInterface>().Verify(x => x.TestMethod(), times);"],

// Should NOT flag Verify with only a failure message (no Times parameter at all)
["new Mock<ISampleInterface>().Verify(x => x.TestMethod(), \"custom failure message\");"],
```

(The `Func<Times>` gap traces to `src/Analyzers/RedundantTimesSpecificationAnalyzer.cs:91-92`, where `HasTimesParameter` requires the parameter type to equal `Moq.Times` exactly. If maintainers decide the analyzer *should* flag the method-group form, file that as a separate enhancement; do not change the analyzer in this issue.)

4b. Add a `[Fact]` for the `using static Moq.Times` form (a data row cannot add a `using` directive to the shared template; verified the diagnostic lands on `AtLeastOnce()`):

```csharp
[Fact]
public async Task ShouldDetectRedundantTimesWithUsingStatic()
{
    const string source = """
using static Moq.Times;

namespace Test
{
    public interface ISampleInterface
    {
        void TestMethod();
    }

    public class TestClass
    {
        public void TestMethod()
        {
            new Mock<ISampleInterface>().Verify(x => x.TestMethod(), {|Moq1420:AtLeastOnce()|});
        }
    }
}
""";

    await Verifier.VerifyAnalyzerAsync(source, ReferenceAssemblyCatalog.Net80WithNewMoq);
}
```

### 5. Moq1200 — `tests/Moq.Analyzers.Test/SetupShouldBeUsedOnlyForOverridableMembersAnalyzerTests.cs`

5a. Extend the dead fixture at line 93 with a static property and a static non-void method (needed because the analyzer only sees the *generic* `Setup<TResult>` overload; see 5b):

```csharp
public class SampleClassWithStaticMembers { public static int StaticField; public const int ConstField = 42; public static readonly int ReadonlyField = 42; public static void StaticMethod() { } public static int StaticProperty { get; set; } public static int GetStaticValue() => 0; }
```

5b. Add these rows to the `@new` list (after line 56; verified against Moq 4.18.4, which is what that list runs):

```csharp
// Static members referenced from a Setup lambda are not overridable and are flagged.
// These bind the generic Setup<TResult>(Expression<Func<T, TResult>>) overload.
["""{|Moq1200:new Mock<SampleClassWithStaticMembers>().Setup(x => SampleClassWithStaticMembers.StaticField)|};"""],
["""{|Moq1200:new Mock<SampleClassWithStaticMembers>().Setup(x => SampleClassWithStaticMembers.ConstField)|};"""],
["""{|Moq1200:new Mock<SampleClassWithStaticMembers>().Setup(x => SampleClassWithStaticMembers.ReadonlyField)|};"""],
["""{|Moq1200:new Mock<SampleClassWithStaticMembers>().Setup(x => SampleClassWithStaticMembers.StaticProperty)|};"""],
["""{|Moq1200:new Mock<SampleClassWithStaticMembers>().Setup(x => SampleClassWithStaticMembers.GetStaticValue())|};"""],

// A static *void* method binds the non-generic Setup(Expression<Action<T>>) overload,
// which IsMoqSetupMethod excludes via its IsGenericMethod check
// (src/Common/ISymbolExtensions.Moq.cs:40-43), so no diagnostic is produced today.
// Known false negative; this row pins the current behavior.
["""new Mock<SampleClassWithStaticMembers>().Setup(x => SampleClassWithStaticMembers.StaticMethod());"""],

// Extension methods cannot be overridden and are flagged.
["""{|Moq1200:new Mock<ISampleInterface>().Setup(x => x.CalculateTwice())|};"""],

// Nested-generic mocked type with an overridable interface member - not flagged.
["""new Mock<IGenericInterface<IGenericInterface<int>>>().Setup(x => x.GetValue());"""],
```

5c. Add the supporting fixture types to the theory's source template (next to the other fixture declarations around lines 89-93):

```csharp
public interface IGenericInterface<T> { T GetValue(); }
public static class SampleExtensions { public static int CalculateTwice(this ISampleInterface s) => s.Calculate(1, 2) * 2; }
```

5d. Add a malformed-lambda test (and `using Microsoft.CodeAnalysis.Testing;` at the top of the file); verified: no diagnostic, no crash:

```csharp
/// <summary>
/// An incomplete Setup lambda (mid-edit code) must not crash the analyzer and
/// produces no diagnostic (current behavior).
/// </summary>
[Theory]
[InlineData("Net80WithOldMoq")]
[InlineData("Net80WithNewMoq")]
public async Task ShouldNotReportOnIncompleteSetupLambda(string referenceAssemblyGroup)
{
    const string source = """
        public interface ISampleInterface
        {
            int Calculate(int a, int b);
        }

        internal class UnitTest
        {
            private void Test()
            {
                new Mock<ISampleInterface>().Setup(x => x.
            }
        }
        """;

    // CompilerDiagnostics.None suppresses CS1001/CS1026/CS1002 from the incomplete lambda.
    await Verifier.VerifyAnalyzerAsync(source, referenceAssemblyGroup, CompilerDiagnostics.None);
}
```

### 6. Moq1100 — `tests/Moq.Analyzers.Test/CallbackSignatureShouldMatchMockedMethodAnalyzerTests.cs`

6a. Add to `validPatterns` (all verified to produce no diagnostic today):

```csharp
// Callbacks chained on setups of *void* methods (ProcessData/ProcessMixed/ProcessReadOnly)
// bind the non-generic Setup(Expression<Action<T>>) overload, which IsMoqSetupMethod
// excludes via its IsGenericMethod check (src/Common/ISymbolExtensions.Moq.cs:40-43),
// so they are not analyzed today - even when the signature mismatches.
// Known false negative; these rows pin the current behavior.
["""new Mock<IFoo>().Setup(m => m.ProcessData(ref It.Ref<string>.IsAny)).Callback((ref string data) => { });"""],
["""new Mock<IFoo>().Setup(m => m.ProcessData(ref It.Ref<string>.IsAny)).Callback((string data) => { });"""],
["""new Mock<IFoo>().Setup(m => m.ProcessMixed(It.IsAny<int>(), ref It.Ref<string>.IsAny, out It.Ref<bool>.IsAny)).Callback((int id, ref string data, out bool success) => { success = true; });"""],
["""new Mock<IFoo>().Setup(m => m.ProcessMixed(It.IsAny<int>(), ref It.Ref<string>.IsAny, out It.Ref<bool>.IsAny)).Callback((int id, string data, bool success) => { });"""],
["""new Mock<IFoo>().Setup(m => m.ProcessReadOnly(in It.Ref<DateTime>.IsAny)).Callback((in DateTime timestamp) => { });"""],
["""new Mock<IFoo>().Setup(m => m.ProcessReadOnly(in It.Ref<DateTime>.IsAny)).Callback((DateTime timestamp) => { });"""],

// TryProcess returns bool, so its setup binds the generic overload and IS analyzed;
// a correct out-parameter callback is not flagged.
["""new Mock<IFoo>().Setup(m => m.TryProcess(out It.Ref<int>.IsAny)).Callback((out int result) => { result = 42; });"""],

// Generic mocked method instantiation with a matching callback
["""new Mock<IFoo>().Setup(x => x.ProcessGeneric<int>(It.IsAny<int>())).Callback((int input) => { });"""],

// Method-group callbacks are not analyzed today (only lambdas and delegate-creation
// expressions are inspected). The second row's parameter type (int) mismatches
// DoWork(string) and is still not flagged - known false negative; pins current behavior.
["""new Mock<IFoo>().Setup(x => x.DoWork("test")).Callback(HandleString);"""],
["""new Mock<IFoo>().Setup(x => x.DoWork("test")).Callback<int>(HandleInt);"""],

// Custom delegate with a matching signature
["""new Mock<IFoo>().Setup(x => x.DoWork("test")).Callback(new CustomStringAction(x => { }));"""],

// Async lambda with zero parameters (zero-parameter callbacks are a valid Moq pattern)
["""new Mock<IFoo>().Setup(x => x.DoWork("test")).Callback(async () => await Task.Delay(1));"""],
```

6b. Add to `invalidPatterns` (all verified, including exact spans):

```csharp
// Out-parameter mismatch on a bool-returning method (missing 'out')
["""new Mock<IFoo>().Setup(m => m.TryProcess(out It.Ref<int>.IsAny)).Callback(({|Moq1100:int result|}) => { });"""],

// Generic mocked method instantiated as int; callback declares string
["""new Mock<IFoo>().Setup(x => x.ProcessGeneric<int>(It.IsAny<int>())).Callback(({|Moq1100:string input|}) => { });"""],

// Custom delegate with a mismatching parameter type (simple and parenthesized lambdas)
["""new Mock<IFoo>().Setup(x => x.DoWork("test")).Callback(new CustomIntAction({|Moq1100:x|} => { }));"""],
["""new Mock<IFoo>().Setup(x => x.DoWork("test")).Callback(new CustomIntAction(({|Moq1100:int x|}) => { }));"""],
```

6c. Supporting fixture changes in the `Template` local function (lines 115-140):

- Add after the `IFoo` interface declaration:

```csharp
public delegate void CustomStringAction(string input);
public delegate void CustomIntAction(int input);
```

- Add inside `TestClass`, before `TestMethod`:

```csharp
private static void HandleString(string input) { }
private static void HandleInt(int input) { }
```

### 7. Cosmetic — `tests/Moq.Analyzers.Test/NoSealedClassMocksAnalyzerTests.cs:171`

Rename the test method `ShoulAnalyzeSealedClassMocks` to `ShouldAnalyzeSealedClassMocks` (typo; no other references exist — verify with a solution-wide search before renaming).

## Validation

1. `dotnet build` — no warnings or errors.
2. `dotnet test --settings ./build/targets/tests/test.runsettings` — all tests pass. Every new row runs 2 (namespaces) x 1-2 (Moq versions) times via the `TestDataExtensions` combinators.
3. Confirm no fixture member in the two previously-dead fixtures remains unreferenced: search the test project for `SampleClassWithStaticMembers`, `ProcessData`, `TryProcess`, `ProcessMixed`, `ProcessReadOnly`, `ProcessGeneric` and check each appears in at least one data row.
4. If any expected-diagnostic markup fails, do **not** change analyzer code; re-check the row against this issue's tables (each expectation here was produced by running the current analyzers).

## Acceptance criteria

- [ ] Moq1300: 8 new data rows (generic/nested-generic interfaces, chained `As`, mock-variable access, delegate type args) + `ShouldNotReportWhenTypeArgumentViolatesClassConstraint` test added; all pass.
- [ ] Moq1400: 5 new `common` rows (ctor-args combos + `MockRepository.Create` pins), 3 new `newMoqOnly` rows (`Mock.Of` predicate), fixture `Foo`/`ISample.Name` added, `ShouldNotCrashOnIncompleteMockCreation` test added; all pass.
- [ ] Moq1202: 3 valid + 3 invalid rows added, `PairChanged`/`ListChanged` events added to both templates, `ShouldAnalyzeRaiseOnMemberAndMethodChainMockAccess` theory added; all pass.
- [ ] Moq1420: 4 new rows (2 `Func<Times>` method-group pins, Times-in-local, message-only) + `ShouldDetectRedundantTimesWithUsingStatic` fact added; all pass.
- [ ] Moq1200: `SampleClassWithStaticMembers` fixture extended and referenced by 6 rows, extension-method and nested-generic rows added, `ShouldNotReportOnIncompleteSetupLambda` test added; all pass.
- [ ] Moq1100: all 5 previously-dead fixture methods referenced by data rows; method-group, custom-delegate, and async-lambda rows added; 4 new invalid rows with exact spans pass.
- [ ] `ShoulAnalyzeSealedClassMocks` renamed to `ShouldAnalyzeSealedClassMocks`.
- [ ] Rows that pin known false negatives carry an explanatory comment (void-Setup exclusion, method-group exclusion, `Func<Times>` parameter type).
- [ ] No analyzer or code-fix source under `src/` is modified.

## Out of scope

- Any behavior change to any analyzer (the false negatives pinned above — void `Setup(Expression<Action<T>>)` exclusion, method-group callbacks, `Func<Times>` overloads, `MockRepository.Create<T>()` — are candidates for follow-up enhancements, filed separately).
- Moq1300 behavior on error types / generic type parameters (companion audit issue changes this; its tests land there).
- Moq1202/Moq1204 rows for non-generic `EventHandler` / custom delegates (companion audit issue "Moq1202/Moq1204 false positives: canonical `Raise` on non-generic `EventHandler` events...").
- Moq1400/Moq1410 rows passing `MockBehavior` via variable or `const` (companion boxed-enum equality issue).
- The two near-duplicate `ArrayExtensionsTests.cs`/`ArrayExtensionTests.cs` files under `tests/Moq.Analyzers.Test/Common/` (tracked by the dead-code audit issue).
- Moq1004 (`NoMockOfLogger`) gaps — minor, and that suite already uses the strongest assertion form.

## Related issues

- #639 / #645 — 100% code-coverage explainer and task epic (this issue advances the "edge cases and error conditions" requirement).
- #687 / #694 — diagnostic message/description coverage epic (complementary; this issue asserts IDs and spans, not message text).
- #904 — Stryker.NET mutation testing (the dead fixtures and same-path/different-semantics gaps here are exactly the class of gap mutation testing surfaces).
